lib/auth.js

```diff
@@ -0,0 +1,70 @@
+module.exports = loopbackAuth;
+
+function loopbackAuth(prefix) {
+  var props = ['accessTokenId', 'currentUserId'],
+      _sessionStorage,
+      _localStorage;
+
+  prefix = (prefix || 'loopback') + ':';
+
+  var auth = {
+    save: save,
+    setUser: setUser,
+    clearUser: clearUser,
+    clearStorage: clearStorage
+  };
+
+  if (typeof window !== 'undefined') {
+    _localStorage = window.localStorage;
```

Comment:
> I don't think storage belongs in this module. All we need to do in the connector itself is ensure the token and any other metadata required by authentication is set on the context and or request.

I agree that browser-specific stuff should be left out of this pull request.

@ritch your proposal may be putting too much burden on the users of this module. So we may end up with including some part of the browser storage in the connector itself. However, even if that happens, then it should be build on top of the non-browser-specific API provided by this PR.

Here is my proposal:

 - In this PR, implement the API proposed by @ritch above
 - Once it is landed, look into ways how to persist and restore the access token in the browser.